### ISSUE TITLE
[BUG] `fuzziness` parameter isn't validated in search queries

### ISSUE THREAD

**Issue by Yury-Fridlyand** [opensearch-project/OpenSearch#4223]
## **Describe the bug**
`fuzziness` parameter value isn't validated.

## **To Reproduce**
```
curl -XGET 'http://localhost:9200/_search?pretty' -d '{"query":{"match_bool_prefix":{"date":{"query":"1972-07-04 00:00:00", "fuzziness":"+++"}}}}' -H 'Content-Type: application/json'
```
returns **200**
Other values tested: `ololo`, `42`, `5,1`, `AUTO:` and so on. Tested in OpenSearch API and other search queries as well.

## **Expected behavior**
An error message, like negative `boost` value produces

## **Plugins**
Default set

## **Host/Environment (please complete the following information):**
 - OS: Win 11 + WSL + docker
 - Version: 2.1.0

## **Additional context**
See also: #3749

**Comment by macohen** [comment]
Assigning to me for now for tracking and investigation.

**Comment by macohen** [comment]
According to the docs, https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#run-fuzzy-queries, valid input is AUTO, 0, 1, 2, or some other positive integer. In https://github.com/opensearch-project/OpenSearch/blob/10bff0c9f5b9ca78b3dc50f5c704dabf41b9d535/server/src/main/java/org/opensearch/common/unit/Fuzziness.java, it looks like we are not checking for that valid input in any constructors.

**Comment by Yury-Fridlyand** [comment]
Exactly.
I see from [elastic documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/common-options.html#fuzziness), `AUTO:int,int` (e.g. `AUTO:1,2`) is also a valid value.

**Comment by vijal-patel** [comment]
could I work on this one?

**Comment by Manishpandey11** [comment]
Hi ,I want to work on this issue

**Comment by dbwiddis** [comment]
@Manishpandey11 Go ahead and submit a PR!  Let us know if you need help.

**Comment by vijal-patel** [comment]
@Manishpandey11, you can go for it, I haven't done much work on this as I don't have the electricity supply to work on this one right now (rolling power cuts in South Africa).

**Comment by subbudvk** [comment]
> According to the docs, https://opensearch.org/docs/latest/opensearch/query-dsl/full-text/#run-fuzzy-queries, valid input is AUTO, 0, 1, 2, or some other positive integer. In https://github.com/opensearch-project/OpenSearch/blob/10bff0c9f5b9ca78b3dc50f5c704dabf41b9d535/server/src/main/java/org/opensearch/common/unit/Fuzziness.java, it looks like we are not checking for that valid input in any constructors.

@macohen : I've raised a MR for this defect. Can you please review this?

**Comment by subbudvk** [comment]
I have written tests for this. But I see some of the existing tests like _testNumerics()_ try to assert double fuzziness values like "1.000" which will fail after the current validation in the constructor. If floating point value is expected, I see we are trying to parse with _Integer.parseInt()_ in _parseCustomAuto()_, so this should already throw NumberFormatException.

**Comment by noCharger** [comment]
Fuzziness interface is refactored https://github.com/opensearch-project/OpenSearch/pull/5433/files

We can determine if any additional validations are required.

**Comment by noCharger** [comment]
From here,
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/test/java/org/opensearch/index/query/FuzzyQueryBuilderTests.java#L118-L119

This case below should return 400 instead of 200. Next step is to fix the bug.


> ## **Describe the bug**
> `fuzziness` parameter value isn't validated.
> 
> ## **To Reproduce**
> Steps to reproduce the behavior:
> 
> ```
> curl -XGET 'http://localhost:9200/_search?pretty' -d '{"query":{"match_bool_prefix":{"date":{"query":"1972-07-04 00:00:00", "fuzziness":"+++"}}}}' -H 'Content-Type: application/json'
> ```
> 
> returns **200** Other values tested: `ololo`, `42`, `5,1`, `AUTO:` and so on. Tested in OpenSearch API and other search queries as well.
> 
> ## **Expected behavior**
> An error message, like negative `boost` value produces
> 
> ## **Plugins**
> Default set
> 
> ## **Host/Environment (please complete the following information):**
> * OS: Win 11 + WSL + docker
> * Version: 2.1.0
> 
> ## **Additional context**
> See also: #3749

**Comment by sejli** [comment]
Closing, fixed in #5805.
